### PR TITLE
Fix bug in attribute type conversion wiith compound types - merge to 1.10

### PR DESCRIPTION
**Pull request by fortnern**
* Fix bug in attribute type conversion wiith compound types (#2016)

* Fix bug in attribute type conversion where the background buffer would
not be initialized with the destination contents when necessary.  Other
minor simplification.

* Committing clang-format changes

* Fix warnings.

* Address review comments.

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* Add note to RELEASE.txt for GitHub #2016 (#2030)

* Fix bug in attribute type conversion where the background buffer would
not be initialized with the destination contents when necessary.  Other
minor simplification.

* Committing clang-format changes

* Fix warnings.

* Address review comments.

* Add RELEASE.txt note for PR #2016

* Add GitHub number to release note

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

### CODE REVIEW

**Review by byrnHDF — APPROVED** [review]

File: src/H5Aint.c
Comment (bmribler) [inline src/H5Aint.c]: Does (dst_type_size * nelmts) need to be checked against buf_size?

File: src/H5Aint.c
Comment (bmribler) [inline src/H5Aint.c]: H5E_NOSPACE here...

File: src/H5Aint.c
Comment (bmribler) [inline src/H5Aint.c:776]: but H5E_CANTALLOC here...

File: src/H5Aint.c
Comment (bmribler) [inline src/H5Aint.c:773]: Very very minor: should "()" be used for consistency as (dst_type_size * nelmts) on line 652 or remove "()" there?

File: test/dtypes.c
Comment (bmribler) [inline test/dtypes.c]: Incorrect error message

File: src/H5Aint.c
Comment (fortnern) [reply src/H5Aint.c]: We could put in an assertion, but it should never be smaller than buf_size, since buf_size is set to nelmts * MAX(src_type_size, dst_type_size);

File: src/H5Aint.c
Comment (fortnern) [reply src/H5Aint.c:773]: We could remove the parens in the other place.

File: src/H5Aint.c
Comment (fortnern) [reply src/H5Aint.c]: This was kept in place from old code but I can change it if we decide to make the other minor changes. Keep in mind making changes here will require they first be made in develop then merged to 1.12 then merged here. In general it's best to closely review the original PR to develop instead of downstream merges :)

**Comment by fortnern** [comment]
I'll be pushing a fix for Binh-Minh's comments through develop and 1.12, hold off on merging this until it makes it here

**Comment by fortnern** [comment]
@bmribler can you take a look at https://github.com/HDFGroup/hdf5/pull/2073 and make sure it addresses your concerns? Thanks

**Comment by fortnern** [comment]
Ok I've just added a commit that addresses @bmribler 's concerns. This should be ready to merge once the tests pass and we get another approval.

**Review by lrknox — APPROVED** [review]

**Pull request merged**
